asciigraf/asciigraf.py

```diff
@@ -41,52 +47,107 @@ def graph_from_ascii(network_string):
     node_char_to_node = map_text_chars_to_text(nodes)
     label_char_to_label = map_text_chars_to_text(labels)
 
-    edge_char_to_edge_map = {}  # {Point -> {"points": [], "nodes": []}}
-    edges = []
-    for pos, char in edge_chars.items():
-        trailing_offset, leading_offset = EDGE_CHAR_NEIGHBOURS[char]
-        neighbor = pos + trailing_offset
-        neighbor_2 = pos + leading_offset
-
-        # we can skip this position if a previous iteration already
-        # mapped the character to an edge
-        if pos not in edge_char_to_edge_map:
-            if neighbor in edge_char_to_edge_map:  # Add this node to the edge
-                edge_char_to_edge_map[pos] = edge_char_to_edge_map[neighbor]
-                edge_char_to_edge_map[pos]["points"].append(pos)
-            else:  # Make a new edge
-                edge_char_to_edge_map[pos] = dict(points=[pos], nodes=[])
-                edges.append(edge_char_to_edge_map[pos])
-
-            # we can look ahead at other neighbor and add it too -- this
-            # step allows us to solve a few extra corner types
-            if (
-                neighbor_2 not in edge_char_to_edge_map
-                and neighbor_2 in edge_chars
-            ):
-                edge_char_to_edge_map[neighbor_2] = edge_char_to_edge_map[pos]
-                edge_char_to_edge_map[pos]["points"].append(neighbor_2)
-
-        neighboring_nodes = [
-            node_char_to_node[pos+pos_offset]
-            for pos_offset in EDGE_CHAR_NEIGHBOURS[char]
-            if pos+pos_offset in node_char_to_node
-        ]
-        edge_char_to_edge_map[pos]["nodes"] += neighboring_nodes
-
-    # Make the networkx graph
-    for edge in edges:
-        edge_map = {pos: edge_chars[pos] for pos in edge["points"]}
-        if len(edge['nodes']) > 2:
-            raise InvalidEdgeError(
-                "Too many nodes:\nNetwork String:\n" +
-                "{}\nAffected Edge:\n{}".format(network_string, draw(edge_map))
-            )
-        elif len(edge['nodes']) < 2:
+    def get_abutting_edge_chars(pos):
```

Comment:
Do these inner-functions need to be inner functions? `graph_from_ascii` might read easier if these weren't in here.

Reply:
I've been struggling with how to resolve this, because `get_abutting_edge_chars` would need 3 additional inputs in order to make it not an inner function, and so would the `build_edge_from_position` function.